pkg/apis/k0sctl.k0sproject.io/v1beta1/cluster/host.go

```diff
@@ -620,6 +620,160 @@ func (h *Host) WaitKubeAPIReady(port int) error {
 	return h.WaitHTTPStatus(fmt.Sprintf("https://localhost:%d/version", port), 200, 401)
 }
 
+// Convert eventTime obtained by kubectl get events to time.Time object.
+func (h *Host) ConvertEventTimeToTimeObject(inputTimestamp string) (time.Time, error) {
+	// Format of eventTime to be obtained with kubectl get events
+	layouts := []string{
+		"2006-01-02T15:04:05Z",
+		"2006-01-02T15:04:05.999999Z",
+	}
+
+	// Convert eventTime to a time object.
+	var t time.Time
+	var err error
+	for _, layout := range layouts {
+		t, err = time.Parse(layout, inputTimestamp)
+		if err == nil {
+			break
+		}
+	}
+
+	return t, err
+}
+
+type KubeSystemEvent struct {
+	Items []struct {
+		Reason    string `json:"reason"`
+		EventTime string `json:"eventTime"`
```

Comment:
Did you try with just time.Time? I'd expect it to work as these kube jsons should come straight from go's json.Marshal